Certificate import ambiguous value of file-name

**Describe the bug**
I would like to import a root certificate and I'm getting the following error:
POST 'http://192.168.88.1:80/rest/certificate/import' returned response code: 400, message: 'Bad Request', details: 'ambiguous value of file-name, more than one possible value matches input'

Despite the API error, the certificate is correctly imported into the RouterOS.
```
[admin@MikroTik] > /certificate/ print 
Flags: L - CRL; T - TRUSTED
Columns: NAME, COMMON-NAME, SKID
#    NAME        COMMON-NAME                                 SKID                                    
0 LT Quad9-root  DigiCert Global G3 TLS ECC SHA384 2020 CA1  8a23eb9e6bd7f9375df96d2139769aa167de10a8
```

**To Reproduce**
```terraform
data "routeros_x509" "quad9_root" {
  data = <<EOT
-----BEGIN CERTIFICATE-----
MIIDeTCCAv+gAwIBAgIQDCJUGB1t/fpm4mTjwXpIvDAKBggqhkjOPQQDAzBhMQsw
CQYDVQQGEwJVUzEVMBMGA1UEChMMRGlnaUNlcnQgSW5jMRkwFwYDVQQLExB3d3cu
ZGlnaWNlcnQuY29tMSAwHgYDVQQDExdEaWdpQ2VydCBHbG9iYWwgUm9vdCBHMzAe
Fw0yMTAzMzAwMDAwMDBaFw0zMTAzMjkyMzU5NTlaMFkxCzAJBgNVBAYTAlVTMRUw
EwYDVQQKEwxEaWdpQ2VydCBJbmMxMzAxBgNVBAMTKkRpZ2lDZXJ0IEdsb2JhbCBH
MyBUTFMgRUNDIFNIQTM4NCAyMDIwIENBMTB2MBAGByqGSM49AgEGBSuBBAAiA2IA
BHipnHWuiF1jpK1dhtgQSdavklljQyOF9EhlMM1KNJWmDj7ZfAjXVwUoSJ4Lq+vC
05ae7UXSi4rOAUsXQ+Fzz21zSDTcAEYJtVZUyV96xxMH0GwYF2zK28cLJlYujQf1
Z6OCAYIwggF+MBIGA1UdEwEB/wQIMAYBAf8CAQAwHQYDVR0OBBYEFIoj655r1/k3
XfltITl2mqFn3hCoMB8GA1UdIwQYMBaAFLPbSKT5ocXYrjZBzBFjaWIpvEvGMA4G
A1UdDwEB/wQEAwIBhjAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwdgYI
KwYBBQUHAQEEajBoMCQGCCsGAQUFBzABhhhodHRwOi8vb2NzcC5kaWdpY2VydC5j
b20wQAYIKwYBBQUHMAKGNGh0dHA6Ly9jYWNlcnRzLmRpZ2ljZXJ0LmNvbS9EaWdp
Q2VydEdsb2JhbFJvb3RHMy5jcnQwQgYDVR0fBDswOTA3oDWgM4YxaHR0cDovL2Ny
bDMuZGlnaWNlcnQuY29tL0RpZ2lDZXJ0R2xvYmFsUm9vdEczLmNybDA9BgNVHSAE
NjA0MAsGCWCGSAGG/WwCATAHBgVngQwBATAIBgZngQwBAgEwCAYGZ4EMAQICMAgG
BmeBDAECAzAKBggqhkjOPQQDAwNoADBlAjAdIuJYQITJrIeX+XKyNUbQGHZ6YhRz
fZrAkUOo847MAe232GlXYw8cE5Vh7hOotCsCMQDgwJgM+AcC2kL4BM7eaaprMi5u
FEMbiAcHmsTBJS3vzvx7eJ9JlMhlrGSu7QrmeIg=
-----END CERTIFICATE-----
EOT
}

resource "routeros_file" "quad9_root" {
  name = "DigiCertGlobalG3TLSECCSHA3842020CA1-1.crt.pem"
  contents = data.routeros_x509.quad9_root.pem
}

resource "routeros_system_certificate" "quad9_root" {
  name        = "Quad9-root"
  common_name = data.routeros_x509.quad9_root.common_name
  import {
    cert_file_name = routeros_file.quad9_root.name
  }
  depends_on = [routeros_file.quad9_root]
}
```

**Expected behavior**
The certificate is uploaded and imported without errors.

**Debug Information**
```
routeros_system_certificate.quad9_root: Creating...
2024-07-29T20:36:03.647+0200 [INFO]  Starting apply for routeros_system_certificate.quad9_root
2024-07-29T20:36:03.649+0200 [DEBUG] routeros_system_certificate.quad9_root: applying the planned Create change
2024-07-29T20:36:03.653+0200 [INFO]  provider.terraform-provider-routeros_v1.57.0: 2024/07/29 20:36:03 [DEBUG] setting computed for "key_usage" from ComputedKeys: timestamp="2024-07-29T20:36:03.653+0200"
2024-07-29T20:36:03.655+0200 [DEBUG] provider.terraform-provider-routeros_v1.57.0: request body:  {"file-name":"DigiCertGlobalG3TLSECCSHA3842020CA1-1.crt.pem","name":"Quad9-root","passphrase":""}: tf_provider_addr=terraform-routeros/routeros tf_req_id=c67913d0-2761-77bd-7683-6c1dbd7a0fe3 tf_rpc=Configure @caller=github.com/terraform-routeros/terraform-provider-routeros/routeros/log.go:27 @module=routeros timestamp="2024-07-29T20:36:03.654+0200"
2024-07-29T20:36:03.655+0200 [DEBUG] provider.terraform-provider-routeros_v1.57.0: POST request URL:  http://192.168.88.1:80/rest/certificate/import: tf_rpc=Configure @caller=github.com/terraform-routeros/terraform-provider-routeros/routeros/log.go:27 @module=routeros tf_provider_addr=terraform-routeros/routeros tf_req_id=c67913d0-2761-77bd-7683-6c1dbd7a0fe3 timestamp="2024-07-29T20:36:03.655+0200"
2024-07-29T20:36:03.666+0200 [DEBUG] provider.terraform-provider-routeros_v1.57.0: response body: [{"certificates-imported":"0","decryption-failures":"0","files-imported":"0","keys-with-no-certificate":"0","private-keys-imported":"0"}]: @caller=github.com/terraform-routeros/terraform-provider-routeros/routeros/log.go:27 @module=routeros tf_provider_addr=terraform-routeros/routeros tf_req_id=c67913d0-2761-77bd-7683-6c1dbd7a0fe3 tf_rpc=Configure timestamp="2024-07-29T20:36:03.666+0200"
2024-07-29T20:36:03.666+0200 [DEBUG] provider.terraform-provider-routeros_v1.57.0: request body:  {"file-name":"","name":"Quad9-root","passphrase":""}: @caller=github.com/terraform-routeros/terraform-provider-routeros/routeros/log.go:27 @module=routeros tf_provider_addr=terraform-routeros/routeros tf_req_id=c67913d0-2761-77bd-7683-6c1dbd7a0fe3 tf_rpc=Configure timestamp="2024-07-29T20:36:03.666+0200"
2024-07-29T20:36:03.666+0200 [DEBUG] provider.terraform-provider-routeros_v1.57.0: POST request URL:  http://192.168.88.1:80/rest/certificate/import: tf_rpc=Configure @caller=github.com/terraform-routeros/terraform-provider-routeros/routeros/log.go:27 @module=routeros tf_provider_addr=terraform-routeros/routeros tf_req_id=c67913d0-2761-77bd-7683-6c1dbd7a0fe3 timestamp="2024-07-29T20:36:03.666+0200"
2024-07-29T20:36:03.670+0200 [DEBUG] provider.terraform-provider-routeros_v1.57.0: error response body:
{"detail":"ambiguous value of file-name, more than one possible value matches input","error":400,"message":"Bad Request"}: tf_provider_addr=terraform-routeros/routeros tf_req_id=c67913d0-2761-77bd-7683-6c1dbd7a0fe3 tf_rpc=Configure @caller=github.com/terraform-routeros/terraform-provider-routeros/routeros/log.go:27 @module=routeros timestamp="2024-07-29T20:36:03.669+0200"
2024-07-29T20:36:03.670+0200 [ERROR] provider.terraform-provider-routeros_v1.57.0: Response contains error diagnostic: diagnostic_detail="" tf_provider_addr=terraform-routeros/routeros tf_req_id=cf691a01-7f40-3e03-dc42-0f4ba83203b7 tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/diag/diagnostics.go:58 @module=sdk.proto tf_resource_type=routeros_system_certificate diagnostic_severity=ERROR diagnostic_summary="POST 'http://192.168.88.1:80/rest/certificate/import' returned response code: 400, message: 'Bad Request', details: 'ambiguous value of file-name, more than one possible value matches input'" tf_proto_version=5.6 timestamp="2024-07-29T20:36:03.670+0200"
2024-07-29T20:36:03.682+0200 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2024-07-29T20:36:03.682+0200 [ERROR] vertex "routeros_system_certificate.quad9_root" error: POST 'http://192.168.88.1:80/rest/certificate/import' returned response code: 400, message: 'Bad Request', details: 'ambiguous value of file-name, more than one possible value matches input'
```

**Additional context**
RouterOS version: 7.15.3
terraform-provider-routeros version: 1.57.0

--- a/routeros/resource_system_certificate.go
+++ b/routeros/resource_system_certificate.go
@@ -70,10 +70,11 @@ func ResourceSystemCertificate() *schema.Resource {
 			Description: "A challenge password for scep client.",
 		},
 		"common_name": {
-			Type:        schema.TypeString,
-			Required:    true,
-			ForceNew:    true,
-			Description: "Common Name (e.g. server FQDN or YOUR name).",
+			Type:             schema.TypeString,
+			Required:         true,
+			ForceNew:         true,
+			Description:      "Common Name (e.g. server FQDN or YOUR name).",
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
 		},
 		"copy_from": {
 			Type:        schema.TypeString,
@@ -82,22 +83,23 @@ func ResourceSystemCertificate() *schema.Resource {
 			Description: "",
 		},
 		"country": {
-			Type:         schema.TypeString,
-			Optional:     true,
-			ForceNew:     true,
-			Description:  "Country Name (2 letter code).",
-			ValidateFunc: validation.StringLenBetween(2, 2),
+			Type:             schema.TypeString,
+			Optional:         true,
+			ForceNew:         true,
+			Description:      "Country Name (2 letter code).",
+			ValidateFunc:     validation.StringLenBetween(2, 2),
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
 		},
 		"crl": {
 			Type:     schema.TypeString,
 			Computed: true,
 		},
 		"days_valid": {
-			Type:        schema.TypeInt,
-			Optional:    true,
-			Computed:    true,
-			ForceNew:    true,
-			Description: "Certificate lifetime.",
+			Type:             schema.TypeInt,
+			Optional:         true,
+			ForceNew:         true,
+			Description:      "Certificate lifetime.",
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
 		},
 		"dsa": {
 			Type:     schema.TypeBool,
@@ -179,7 +181,6 @@ func ResourceSystemCertificate() *schema.Resource {
 		"key_usage": {
 			Type:        schema.TypeSet,
 			Optional:    true,
-			Computed:    true,
 			ForceNew:    true,
 			Description: "Detailed key usage descriptions can be found in RFC 5280.",
 			Elem: &schema.Schema{
@@ -207,19 +208,22 @@ func ResourceSystemCertificate() *schema.Resource {
 						"tls-server",
 						"tls-client",
 					}, false)},
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
 		},
 		"locality": {
-			Type:        schema.TypeString,
-			Optional:    true,
-			ForceNew:    true,
-			Description: "Locality Name (eg, city).",
+			Type:             schema.TypeString,
+			Optional:         true,
+			ForceNew:         true,
+			Description:      "Locality Name (eg, city).",
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
 		},
 		KeyName: PropName("Name of the certificate. Name can be edited."),
 		"organization": {
-			Type:        schema.TypeString,
-			Optional:    true,
-			ForceNew:    true,
-			Description: "Organizational Unit Name (eg, section)",
+			Type:             schema.TypeString,
+			Optional:         true,
+			ForceNew:         true,
+			Description:      "Organizational Unit Name (eg, section)",
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
 		},
 		"private_key": {
 			Type:     schema.TypeBool,
@@ -327,10 +331,11 @@ func ResourceSystemCertificate() *schema.Resource {
 			Computed: true,
 		},
 		"state": {
-			Type:        schema.TypeString,
-			Optional:    true,
-			ForceNew:    true,
-			Description: "State or Province Name (full name).",
+			Type:             schema.TypeString,
+			Optional:         true,
+			ForceNew:         true,
+			Description:      "State or Province Name (full name).",
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
 		},
 		"status": {
 			Type:        schema.TypeString,
@@ -338,22 +343,24 @@ func ResourceSystemCertificate() *schema.Resource {
 			Description: "Shows current status of scep client.",
 		},
 		"subject_alt_name": {
-			Type:        schema.TypeString,
-			Optional:    true,
-			ForceNew:    true,
-			Description: "SANs (subject alternative names).",
+			Type:             schema.TypeString,
+			Optional:         true,
+			ForceNew:         true,
+			Description:      "SANs (subject alternative names).",
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
 		},
 		"trusted": {
 			Type:             schema.TypeBool,
 			Optional:         true,
-			DiffSuppressFunc: AlwaysPresentNotUserProvided,
 			Description:      "If set to yes certificate is included 'in trusted certificate chain'.",
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
 		},
 		"unit": {
-			Type:        schema.TypeString,
-			Optional:    true,
-			ForceNew:    true,
-			Description: "Organizational Unit Name (eg, section).",
+			Type:             schema.TypeString,
+			Optional:         true,
+			ForceNew:         true,
+			Description:      "Organizational Unit Name (eg, section).",
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
 		},
 	}
 
@@ -377,7 +384,7 @@ func ResourceSystemCertificate() *schema.Resource {
 			return diag.FromErr(err)
 		}
 
-		if keyFile, ok := bl["key_file_name"]; ok {
+		if keyFile, ok := bl["key_file_name"]; ok && keyFile.(string) != "" {
 			params = MikrotikItem{KeyName: d.Get(KeyName).(string), "file-name": keyFile.(string)}
 			if passwd, ok := bl["passphrase"]; ok {
 				params["passphrase"] = passwd.(string)

